mypy/checkexpr.py

```diff
@@ -1502,6 +1521,19 @@ def not_ready_callback(self, name: str, context: Context) -> None:
         self.chk.handle_cannot_determine_type(name, context)
 
 
+# TODO: What's a good name for this function?
+def has_unfinished_types(t: Type) -> bool:
```

Comment:
There is a similar function already in `checker.py`: `is_valid_inferred_type`. You can probably use it instead of this.